platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/topology/StaticTopology.java

```diff
@@ -17,78 +17,64 @@
-import com.swirlds.platform.system.address.AddressBook;
 import edu.umd.cs.findbugs.annotations.NonNull;
-import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
-import java.util.Objects;
+import java.util.Map;
 import java.util.Random;
-import java.util.function.Predicate;
+import java.util.Set;
 
 /**
  * A bidirectional topology that never changes.
  */
 public class StaticTopology implements NetworkTopology {
     private static final long SEED = 0;
 
-    private final NodeId selfId;
+    /** nodes are mapped so lookups are efficient. **/
+    private Map<NodeId, Long> peerNodeToIdMap = new HashMap<>();
+
     /**
-     * Two nodes are neighbors if their indexes in the address book are neighbors in the connection graph.
+     * Two nodes are neighbors if their node indexes are neighbors in the connection graph.
      */
-    private final AddressBook addressBook;
-
     private final RandomGraph connectionGraph;
 
+    private final NodeId selfId;
+
     /**
      * Constructor.
      *
      * @param random            a source of randomness, used to chose random neighbors, does not need to be
      *                          cryptographically secure
-     * @param addressBook       the current address book
+     * @param peers             the set of peers in the network
      * @param selfId            the ID of this node
      * @param numberOfNeighbors the number of neighbors each node should have
      */
     public StaticTopology(
             @NonNull final Random random,
-            @NonNull final AddressBook addressBook,
+            @NonNull final List<PeerInfo> peers,
             @NonNull final NodeId selfId,
             final int numberOfNeighbors) {
-        this.addressBook = Objects.requireNonNull(addressBook, "addressBook must not be null");
-        this.selfId = Objects.requireNonNull(selfId, "selfId must not be null");
-        this.connectionGraph = new RandomGraph(random, addressBook.getSize(), numberOfNeighbors, SEED);
-
-        if (!addressBook.contains(selfId)) {
-            throw new IllegalArgumentException("Address book does not contain selfId");
-        }
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public List<NodeId> getNeighbors() {
-        return getNeighbors((nodeId -> true));
+        this.peerNodeToIdMap = map(peers);
+        this.selfId = selfId;
+        this.connectionGraph = new RandomGraph(random, peers.size() + 1, numberOfNeighbors, SEED);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public List<NodeId> getNeighbors(final Predicate<NodeId> filter) {
-        final int selfIndex = addressBook.getIndexOfNodeId(selfId);
-        return Arrays.stream(connectionGraph.getNeighbors(selfIndex))
-                .mapToObj(addressBook::getNodeId)
-                .filter(filter)
-                .toList();
+    public Set<NodeId> getNeighbors() {
+        return peerNodeToIdMap.keySet();
```

Comment:
In the previous implementation this method called the predicate argument method.  That method has been deleted.  In the implementation of that deleted method, the definition of Neighbor was scoped to the `connectIonGraph` which is a `RandomGraph` that also models the number of allowed degrees or edges between nodes.  If there are 100 nodes and each node has only 10 degrees in the graph then each node has 10 neighbors.   The old implementation would provide the 10 neighbors a node is allowed to talk to.  This implementation returns all neighbors as being allowed to talk to.

Reply:
Discussed with @edward-swirldslabs. The `StaticTopology` implementation, given nodes ranging from 1-100 returned what's equivalent to the 'peersList` (i.e. the list of peers), all the time. There's a specific test case for this. Agreed it did that based on what's returned by `connectionGraph.getNeighbors()`. I understand there's been some historic discussions around what the future of this class should look like, but the goal of this PR is to introduce an equivalent logic that does not use the AB. 

Happy to discuss alternatives/limitations.

Reply:
This how now been implemented as a fully connected network, eliminating the use of RandomGraph completely. Resolving.